engine/src/main/java/org/terasology/world/chunks/internal/ChunkImpl.java

```diff
@@ -446,6 +446,21 @@ public int hashCode() {
     }
 
     @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof ChunkImpl)) {
+            return false;
+        }
+
+        ChunkImpl that = (ChunkImpl) obj;
+
+        return Objects.equal(this.chunkPos, that.chunkPos);
```

Comment:
This one makes me wonder. Under regular circumstances you would indeed expect two chunks at the same location to equal each other. But are we setting ourselves up for potential bugs here?

Two cases come to mind:

* We have a known issue about duplicate chunks being loaded (#2590) - could this somehow affect that? If somehow a dupe chunk loads but a system interaction changes only one of them. I don't know if dupe chunks at all persist beyond a timing issue or something causing two load requests to be issued.
* In the future could somebody come up with a way where we could have two *valid* chunks in the same location? Imagine something related to dimensions somehow, either completely separate worlds (why anybody would compare two such chunks I don't know .. but MC does relate the position of Nether chunks vs Overworld chunks so who knows) or in-place chunks where a weird bizarro world shows two different players different chunks. Or phasing maybe (like bits of world with progression locked to individual players)

Perhaps we should add in hashing some additional stuff from each chunk and compare those as well? I don't know how frequently this method is likely used (or really a parent implementation)